### ISSUE TITLE
BR1 - No indication when banner progresses, for screen reader

### ISSUE THREAD

**Issue by jszasz** [brharp/hjckrrh#118]
eg CCS site.

If you use a screen reader and keyboard controls, you 'tab' to the link "CCS News" (which has been displayed for 3 seconds already) and hear that spoken to you (one second time). You wait one second (in this time, the banner progresses to the next slide, Google Apps) then you press "enter" to go to the link, and you are taken to the Google Apps page, when you thought you were going to the CCS News Page.

6 seconds is plenty of time to click a link or not, or to view the banner. But if a user tabs to the link text half way through the six seconds, they don't have enough time to hear the link and react to it.

**Comment by jszasz** [comment]
Could we make it so that the slideshow pauses when it receives focus?

**Comment by mmafe** [comment]
Fixed so the banner pauses the moment you focus on the (linked) title. It also pauses when you use the next and previous slide button.

**Comment by jpinsent** [comment]
The fix is working as intended, but this issue is still present.

Screen reader users who haven't paused the slideshow are still able to listen to the image alt text, have the banner automatically roll over to the next image without any prompt, and then go down and hear the linked title for the wrong slide. This makes it seem like the alt text for the previous banner is associated with the wrong linked title.

By listening to the image alt text, waiting for the slideshow to progress to the next slide, and then going down and listening to the title, it's possible to have the screen reader read the following:
```
banner landmark 
button
 Pause
graphic
Glowing warm ariel image of Johnston Hall and surrounding trees
link
Wilson Street Promenade
```
The alt text for the image associated with the Wilson Street Promenade should be "Ariel image of people walking on a green grass pathway and trees on a city street", and the linked title associated with the "Glowing warm ariel image of Johnston Hall and surrounding trees" alt text should be "Johnston Hall".

Maybe we should change the focus order of the title so that it occurs before the image? This would force the slideshow to pause before the user gets to the image. The issue would still occur if the user attempted to go backwards through the banner though.

We could also have it so that any of the controls receiving focus pauses the autoplay, which would force the user to pause the slideshow before getting to the image. The issue would still occur if the user navigated the page via graphics while the banner was on autoplay and then tried to read the title. Not sure if we need our fix to be that exhaustive, though.

The simplest fix would be if we could have the slideshow pause while the screen reader is reading the image alt text, but the image doesn't appear to be focusable, so I'm not sure if it can be done.

**Comment by jpinsent** [comment]
The banner has been updated so that when any component within the banner is focused upon, the slideshow pauses. If the banner is playing and any component within the banner currently has focus, the heading and summary text are read aloud through the screen reader whenever the banner progresses. The alt text is not read by the screen reader unless the user navigates to it.

This issue has been resolved for the most part, however, there is still one situation in which the user can encounter it. If someone with a screen reader navigates to the banner graphic, as long as a component within the banner is not focused upon, they will not hear the banner change through the screen reader. They can still navigate down to the title of the next slide after it rolls over and have the incorrect image alt text associated with the wrong title.

This is a very specific circumstance, so I'll log a separate normal priority issue for it and close this issue.

**Comment by jpinsent** [comment]
The following issues have come up as a result of this issue:
- #211 - Image alt text is not read when the banner progresses (Normal)
- #210 - Banner change is not announced when focus is outside of the banner and the user navigates to the banner image(Normal)
- #209 - NVDA Chrome reads headlines twice and in a random order (Normal)
- #208 - NVDA Firefox does not space the slideshow play button text properly (Normal)
- #207 - Play/Pause button doesn't work properly when clicked out of focus (**High**)
